var/spack/repos/builtin/packages/libflame/package.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libflame(AutotoolsPackage):
+    """libflame is a portable library for dense matrix computations,
+    providing much of the functionality present in LAPACK, developed
+    by current and former members of the Science of High-Performance
+    Computing (SHPC) group in the Institute for Computational
+    Engineering and Sciences at The University of Texas at Austin.
+    libflame includes a compatibility layer, lapack2flame, which
+    includes a complete LAPACK implementation."""
+
+    homepage = "https://www.cs.utexas.edu/~flame/web/libFLAME.html"
+    url      = "https://github.com/flame/libflame/archive/5.1.0.tar.gz"
+    git      = "https://github.com/flame/libflame.git"
+
+    version('develop', branch='master')
+    version('5.1.0', sha256='e7189b750890bd781fe773f366b374518dd1d89a6513d3d6261bf549826384d1')
+
+    provides('lapack', when='+lapack2flame')
+
+    variant('lapack2flame', default=True)
+
+    variant('threads', default='no',
+            description='Multithreading support',
+            values=('pthreads', 'openmp', 'no'),
+            multi=False)
+
+    variant('static', default=True,
+            description='Build static library')
+
+    variant('shared', default=True,
+            description='Build shared library')
+
+    variant('debug', default=False,
+            description='Build with debugging support')
+
+    # TODO: Libflame prefers to defer to an external
+    # LAPACK library for small problems. Is this to be
+    # implemented in spack ?
+
+    # There is a known issue with the makefile :
+    # https://groups.google.com/forum/#!topic/libflame-discuss/lQKEfjyudOY
+    patch('Makefile_5.1.0.patch', when='@5.1.0')
+
+    def configure_args(self):
+        config_args = []
+
+        if '+lapack2flame' in self.spec:
+            config_args.append("--enable-lapack2flame")
+
+        if '+static' in self.spec:
+            config_args.append("--enable-static-build")
+
+        if '+shared' in self.spec:
+            config_args.append("--enable-dynamic-build")
+
+        if '+debug' in self.spec:
+            config_args.append("--enable-debug")
```

Comment:
I would explicitly disable these in an else clause.

Reply:
I don't see any explicit disable options to configure in the docs : http://www.cs.utexas.edu/~flame/web/libflame.pdf#subsection.2.3.1

Reply:
`--enable` usually comes with a corresponding `--disable` that isn't always documented.